### DISCUSSION TITLE
CPU recommendation?

### DISCUSSION THREAD

**Discussion by jadasse** [86Box/86Box#1743] [General]
Is there a host CPU anyone would recommended for 86box to stay at 100% no matter what?

**Comment by jadasse** [comment]
Also, does the host cpu's ghz matter for 86box to stay at 100%?

**Reply by ghost** [reply]
As I know, but I hope that others will correct me if I'm not - the % depends on these things:
 - *The choosen CPU complexity*
   The more complex the cpu, the higher chance for lower %. 
   For example K6 can be slower than Pentium, and PMMX is also slower than Pentium, even on the same clock speed.
 - *The choosen CPU's clock speed*
   Of course, higher clock speed needs more instructions emulated per seconds.
   Slower VM CPU speeds probably results in higher %.
 - *The list of selected hardware*
   Based on my experiences the VM, can get slower depending from how much hardware is added.
   Older chipsets, and less hardware means more %.
 - *The host CPU*
   For the host mostly the [single core performance](https://www.cpubenchmark.net/singleThread.html), or IPC matters, and the max clock speed. If you're on a laptop you should turn on performance mode, to use turbo clock speeds.
   
 If I have to give advice, the 486-series PCs are good chance to keep up with 100% even on a 10 year old medium-category host hardware. With an iDX4 + PCI + Voodoo and dynamic recompiler on, you surely can play old 3D games under Windows 95. 
 
 If you have somewhat newer hardware you can safely try Pentium, but you have to choose a clock speed which can provide you constant 95%-100%+. The K6, Pentium II, K6-2 and more recent processors surely need a modern host CPU, with turbo clock.
 
 The main problems usually starts when the basic IBM PC with 8088 at 4.77 MHz can't keep up with 95-100%+, probably there are some configuration problems in these cases in the background.

**Reply by jadasse** [reply]
I meant for host cpus.

**Reply by jadasse** [reply]
I checked and I think 11900kf would be potential

**Reply by ghost** [reply]
Actually I think a benchmark from someone who have such a high-end PC can be the best advisor.

Currently I think these are the hardest emulated configs:
  1, K6-III+ with the 500 MHz option
  2, Via Cyrix III with the 700 MHz option
  3, Any chipsets with lots of integrated hardware (e.g. Via Apollo series) + many many extra hardwares from all the categories

My personal opinion, that maybe these configs with any current hardware can't run at 100%, but I hope that someone can prove me that this statement is wrong. However the 11900KF truly seems to be promising with its 5.3 GHz turbo clock, and it's very good single-core benchmark value.

Maybe a *benchmark VM* could be created with Quake timedemo, so it can be sended to other people with a "watch how the percent moves" messages, who has high-end hardware, but not fluent in emulation, and it can give you more information.

**Reply by jadasse** [reply]
Cool

**Reply by jadasse** [reply]
Although the Ryzen 9 5950X is more advanced than the Intel Core i9 11900KF, the i9 slightly beats 5950X in turbo speed and single-thread performance; other than that, the 5950x may be the recommended CPU, almost DOUBLES the CPU benchmark of 11900KF.

**Reply by jadasse** [reply]
**(THIS IS ALL A THEORY,  I am not a developer, unlike most people that help my dumb curiosity on GitHub.)** Also, the AMD Ryzen Threadripper PRO 3995wx is almost twice the performance of 5950x, with 64 cores. Also, the 3995wx is FOUR times the performance of the 11900kf. However, the clock speed is a mere 2.7 GHz, with a turbo speed of 4.2 GHz. However, in this case, cores do not matter because those benefit in multitasking, which 86Box is not quite in the category since it is a stand-alone program. So, maybe, the more clock speed, the least stuttering 86Box suffers from, which means better performance. So, out of the three, 11900kf, the 5950x, and the 3995wx, those are probably the most advanced in their categories, with 11900kf being the most advanced in Intel Core, Intel recommends in general, and AMD Ryzen (Threadripper) for consumers. So, what does **anyone** out of these three **(or others if willing to say so or know about in the first place, for instance, Xeon)** recommend that has the least amount of 86box stuttering? Overall, (what I think) for anyone building a PC strong enough for 86Box to have minimal slow-downs or stuttering, maybe the factory default clock speed (both base and turbo) have to be high enough for 86Box to work.

**Reply by jadasse** [reply]
hello?

**Reply by TC1995** [reply]
Offtopic: what time is it there? Because not everyone can reply instantly and depending on the timezone.

**Reply by jadasse** [reply]
EDT: 7:56

**Reply by jadasse** [reply]
_(There's also the 11900k)_

**Reply by jadasse** [reply]
You're right, it's 2:03 am in Hungary, my bad

**Reply by jadasse** [reply]
Hello?

**Reply by headsetguy37** [reply]
> Hello?

Please remember to be patient.

**Reply by jadasse** [reply]
> > Hello?
> 
> 
> 
> Please remember to be patient.

_oops_

**Reply by jadasse** [reply]
However, I would also bring up an alternative to the 11900kf/k; there is also the 10900kf/k. I heard that the 11900k is pretty lackluster when it comes to performance. However, the 10900k is not as high in single performance as 11900kf/k or 5950x.

**Reply by jadasse** [reply]
The 12900k/kf CPU just came out, and it has a VERY HIGHER single-core benchmark value than 11900k!!! So, overall, the 12900k/kf may be the new recommended CPU

**Comment by headsetguy37** [comment]
"No matter what" is very broad. If you mean you want to know what CPU can emulate the fastest CPU that 86Box can handle, with no stuttering, I would imagine you'd need a _very_ powerful processor for that. The clock speed ("the GHz") matters, but so, I think, do the amount of cache the CPU has, the number of cores, and the number of threads per core. (I'm no expert on emulation; this is just my two cents. I don't have any specific suggestions at this time.)

**Reply by jadasse** [reply]
Oh ok

**Comment by Olddantrucker** [comment]
How about on the AMD side what's a good one that can run it at or near 100? I have an I78000 and it can run Win 98 SE Pentium II between 98 and 100 during racing games on PCEM using a LIVE CD when copy protection prevents ISO's from playing. I can install said ISO but cannot play it.  Ultra ISO wounded up not making any difference despite the Cue image.

**Reply by jadasse** [reply]
> 

I don’t specifically know, but I like to recommend the most powerful AMD single thread performed processor today, in this case, the Ryzen 9 5950x.

**Reply by jadasse** [reply]
But for intel, the 12900k
